### PR TITLE
Fixes #143 - Upgraded to Hyrax 2.0 beta4

### LINKED ISSUE #143: Update to Hyrax 2.0 beta4
### Descriptive summary

[2.0 Beta 4](https://github.com/samvera/hyrax/commit/a8873d1c3647bbe7a326221192e726de77a76e5d) is released, let's get upgraded to this for horton.

### Rationale

Stay up to date with upstream.

### PR DESCRIPTION
**Pull request by hweng**
Fixes #143 ; refs #143

A list of changes:
- New database migrations
- User notifications
- View changes
- New locales

@ucsdlib/developers - please review

### CODE REVIEW

File: spec/presenters/hyrax/object_resource_presenter_spec.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Generated via
+#  `rails generate hyrax:work ObjectResource`
+require 'rails_helper'
+
+RSpec.describe Hyrax::ObjectResourcePresenter do
```
Comment (mcritchlow) [inline spec/presenters/hyrax/object_resource_presenter_spec.rb]: Do we need this boilerplate test file? Is this an upstream file we've customized locally and need to test?

File: config/initializers/hyrax.rb [hunk 2]
```diff
@@ -22,8 +22,7 @@
   # How many notifications should be displayed on the dashboard
   # config.max_notifications_for_dashboard = 5
 
-  # How often clients should poll for notifications
-   config.notifications_update_poll_interval = 60.seconds
+  config.realtime_notifications = true
```
Comment (mcritchlow) [inline config/initializers/hyrax.rb:25]: Yay!

File: app/presenters/hyrax/object_resource_presenter.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# Generated via
+#  `rails generate hyrax:work ObjectResource`
+module Hyrax
+  class ObjectResourcePresenter < Hyrax::WorkShowPresenter
```
Comment (mcritchlow) [inline app/presenters/hyrax/object_resource_presenter.rb]: Similar to the question for the related test file. I'm wondering about the definition of this (empty) class.

**Review by mcritchlow — COMMENTED** [review]
This looks great to me, thanks @hweng ! Just have a couple questions on some of the additions that others might be interested in.

File: app/indexers/object_resource_indexer.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# Generated via
+#  `rails generate hyrax:work ObjectResource`
+class ObjectResourceIndexer < Hyrax::WorkIndexer
```
Comment (lsitu) [inline app/indexers/object_resource_indexer.rb]: Do we need this boilerplate file?

**Comment by lsitu** [comment]
👍. Just the same concerns regarding the additional boilerplate files as @mcritchlow noted.

File: app/presenters/hyrax/object_resource_presenter.rb (hunk 3 above)
Comment (hweng) [reply app/presenters/hyrax/object_resource_presenter.rb]: Good question!  It was automatically generated when run the task for new locales. I tried before that the tests would pass without these two classes: object_resource_presenter.rb and object_resource_indexer.rb. 
And I put them back to see if you would like to keep these templates or not?

Comment (hweng) [reply app/presenters/hyrax/object_resource_presenter.rb]: About Presenters in rails: 
https://gist.github.com/somebox/5a7ebf56e3236372eec4
http://nithinbekal.com/posts/rails-presenters/

File: app/indexers/object_resource_indexer.rb (hunk 4 above)
Comment (hweng) [reply app/indexers/object_resource_indexer.rb]: Thanks for the comment. Please refer to the above reply as they are the similar question.

File: app/presenters/hyrax/object_resource_presenter.rb (hunk 3 above)
Comment (lsitu) [reply app/presenters/hyrax/object_resource_presenter.rb]: @hweng I think we've got both in horton:
https://github.com/ucsdlib/horton/blob/develop/app/presenters/object_show_presenter.rb
https://github.com/ucsdlib/horton/blob/develop/app/indexers/work_indexer.rb

Comment (hweng) [reply app/presenters/hyrax/object_resource_presenter.rb]: @lsitu Yes, Some naming convention concerns. As the model class is ObjectResource. Do we want to use the same convention for indexer and presenter, or keep the current name?

**Comment by VivianChu** [comment]
👍  looks good to me. I just have the same concerns as @mcritchlow & @lsitu

File: app/presenters/hyrax/object_resource_presenter.rb (hunk 3 above)
Comment (lsitu) [reply app/presenters/hyrax/object_resource_presenter.rb]: @hweng I believe kind of code refactoring is needed if we want to change it. That could be related to https://github.com/ucsdlib/horton/issues/145 when moving forward.

Comment (hweng) [reply app/presenters/hyrax/object_resource_presenter.rb]: @lsitu Sounds good to me. It fine for me to remove that two classes in this pull request.

**Comment by hweng** [comment]
@ucsdlib/developers - It was removed. please review.

**Comment by lsitu** [comment]
@mcritchlow @VivianChu Either remove those boilerplate files for now or refactoring codes to use them will be fine with me. What do you think?

**Comment by VivianChu** [comment]
Either way is fine with me.

**Review by lsitu — APPROVED** [review]
Okay, let's move it forward then.

**Pull request merged**